pkg/rehearse/jobs.go

```diff
@@ -191,6 +222,34 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 	return rehearsals
 }
 
+// hasChangedTemplateVolume is checking if a list of volumes are using a template's configMap that has been changed.
+// It returns also the index of the volumes that has the changed template as well its name too.
+func hasChangedTemplateVolume(volumes []v1.Volume, cmClient coreclientset.ConfigMapInterface, templates map[string]*templateapi.Template) (bool, int, string, error) {
+	for index, volume := range volumes {
+		if volume.VolumeSource.ConfigMap != nil {
+			cm, err := cmClient.Get(volume.VolumeSource.ConfigMap.Name, metav1.GetOptions{})
```

Comment:
I asked this earlier in https://github.com/openshift/ci-operator-prowgen/pull/100#discussion_r262529365 but there was no answer. Why do we need to touch the cluster here? We know which templates changed already, and we should be able to detect which jobs use them from the config. What do we need the cluster for?

The reason I am asking is that until now, `ConfigureRehearsalJobs` was able to run offline, entirely over the config data present in the repository. I'd prefer to keep it that way.